src/main/java/service/AuthorService.java

```diff
@@ -22,8 +22,20 @@ public class AuthorService {
     private ThisIsMyFirstConditionalBean timfcb;
 
     @Transactional
-    public void saveAuthor(Author author) {
+    public Author saveAuthor(Author author) {
+        if (author == null) {
+            throw new IllegalArgumentException("The object can't be null");
+        }
+
+        if (findAuthor(author.getId()).isPresent())
+        {
+            throw new IllegalArgumentException
+                    ("The object with Id " + author.getId() + " already exists in database");
+        }
+
```

Comment:
1. if you want this method only to create new entities, make it so that the parameter is a different class (dto that is mapped to the entity and the id is excluded)
2. in case you proved null value as a parameter to the repository's save method the exception will be thrown